sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala

```diff
@@ -97,9 +97,24 @@ trait CheckAnalysis extends PredicateHelper {
       case InsertIntoStatement(u: UnresolvedRelation, _, _, _, _) =>
         failAnalysis(s"Table not found: ${u.multipartIdentifier.quoted}")
 
-      case u: UnresolvedV2Relation if isView(u.originalNameParts) =>
-        u.failAnalysis(
-          s"Invalid command: '${u.originalNameParts.quoted}' is a view not a table.")
+      case InsertIntoStatement(u: UnresolvedV2Relation, _, _, _, _)
+          if isView(u.originalNameParts) =>
+        u.failAnalysis(s"Invalid command: '${u.originalNameParts.quoted}' is a view not a table.")
+
+      case InsertIntoStatement(u: UnresolvedV2Relation, _, _, _, _) =>
+        failAnalysis(s"Table not found: ${u.originalNameParts.quoted}")
+
+      case statement: ParsedTableStatement if isView(statement.tableName) =>
+        failAnalysis(s"Invalid command: '${statement.tableName.quoted}' is a view not a table.")
+
+      case statement: ParsedTableStatement if isView(statement.tableName) =>
```

Comment:
This case is exactly the same as the previous one. Is it a typo?

Reply:
Yes, it should not have the `if isView` part.